crates/pixi_uv_conversions/src/git_url.rs

```diff
@@ -0,0 +1,149 @@
+use std::str::FromStr;
+use url::Url;
+use uv_pep508::VerbatimUrl;
+use uv_redacted::DisplaySafeUrl;
+
+/// A URL that may have a git+ prefix, with methods to handle both representations
+#[derive(Debug, Clone, PartialEq)]
+pub struct GitUrlWithPrefix {
+    /// The base URL without git+ prefix
+    base_url: Url,
+    /// Whether this URL originally had a git+ prefix
+    has_git_plus: bool,
+}
+
+impl GitUrlWithPrefix {
+    /// Parse a URL that may have git+ prefix
+    pub fn parse(url_str: &str) -> Result<Self, url::ParseError> {
```

Comment:
maybe we can  use directly FromStr instead of having two functions?

Reply:
Yeah, I never know about this because I like the name parse :). But let's do what you suggested for now.